src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -401,15 +401,14 @@ def recover_vault_or_hsm(cmd, client, resource_group_name=None, location=None, v
 
 
 def recover_hsm(cmd, client, hsm_name, resource_group_name, location, no_wait=False):
-    from azure.cli.core._profile import Profile
+    from azure.cli.core._profile import Profile, _TENANT_ID
 
     ManagedHsm = cmd.get_models('ManagedHsm', resource_type=ResourceType.MGMT_KEYVAULT)
     ManagedHsmSku = cmd.get_models('ManagedHsmSku', resource_type=ResourceType.MGMT_KEYVAULT)
 
     # tenantId and sku shouldn't be required
     profile = Profile(cli_ctx=cmd.cli_ctx)
-    _, _, tenant_id = profile.get_login_credentials(
-        resource=cmd.cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
```

Comment:
I suddenly found that `az keyvault recover` supports `--subscription`, but `--subscription` doesn't really work, since its value is stored in `cli_ctx.data['subscription_id']`

https://github.com/Azure/azure-cli/blob/21266e02457263420b62fba1c529f69e28852414/src/azure-cli-core/azure/cli/core/commands/client_factory.py#L70

but is not used when calling `profile.get_login_credentials`.

Reply:
`--subscription` actually works as expected because `az keyvault recover` calls mgmt plane client factory and the client factory takes `cli_ctx.data['subscription_id']` as default subscription: https://github.com/Azure/azure-cli/blob/b527951889402f7bdf79a7782ac6e0c9c9b48139/src/azure-cli-core/azure/cli/core/commands/client_factory.py#L69-L70

Reply:
Even though mgmt-plane client factory is created with the correct subscription, the `tenant_id` retrieved by L411 may be incorrect - the current `tenant_id` may not be the same as the `tenant_id` of `--subscription`.

Reply:
got it! updated.